featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -651,6 +660,28 @@ def _feature_in_relationship_path(self, relationship_path, feature):
 
         return False
 
+    def _get_matching_inputs(self, all_features, entity, max_depth, input_types,
+                             primitive, direct_only=False, feature_filter=None):
```

Comment:
is there a more descriptive name we can use for `direct_only`?

Reply:
What do you think about `require_direct` or `require_direct_input`?

Reply:
I like `require_direct_input`. should we change `need_direct` later to that too?

Reply:
Yeah I think that's the most clear.